<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from pluto.xacro                    | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="pluto">
  <link name="base_footprint">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.001 0.001 0.001"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="base_link">
    <inertial>
      <mass value="50"/>
      <inertia ixx="2.1505333333333336" ixy="0.0" ixz="0.0" iyy="2.1505333333333336" iyz="0.0" izz="2.1505333333333336"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="-0.015 -0.00275 0"/>
      <geometry>
        <!--box size="${base_length} ${base_width} ${base_height}" /-->
        <mesh filename="package://pluto/models/Chassis.dae" scale="1 1 1"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-0.015 -0.00275 0"/>
      <geometry>
        <!--box size="${base_length} ${base_width} ${base_height}"/-->
        <mesh filename="package://pluto/models/Chassis.dae" scale="1 1 1"/>
      </geometry>
    </collision>
  </link>
  <joint name="base_link_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="base_footprint"/>
    <child link="base_link"/>
  </joint>
  <link name="torso">
    <inertial>
      <mass value="20"/>
      <inertia ixx="0.4010744666666667" ixy="0.0" ixz="0.0" iyy="0.45591306666666664" iyz="0.0" izz="0.16021473333333336"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.32"/>
      <geometry>
        <mesh filename="package://pluto/models/Torso.dae" scale="1 1 1"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.32"/>
      <geometry>
        <mesh filename="package://pluto/models/Torso.dae" scale="1 1 1"/>
      </geometry>
    </collision>
  </link>
  <joint name="torso_joint" type="revolute">
    <axis rpy="0 0 0" xyz="0 0 1"/>
    <parent link="base_link"/>
    <child link="torso"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <limit effort="1.0" lower="-1.57079" upper="1.57079" velocity="1.0"/>
  </joint>
  <link name="right_shoulder">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://pluto/models/Shoulder.dae" scale="1 1 1"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://pluto/models/Shoulder.dae" scale="1 1 1"/>
      </geometry>
      <surface>
        <friction>
          <ode>
            <mu>0.01</mu>
            <mu2>0.01</mu2>
          </ode>
        </friction>
      </surface>
    </collision>
    <inertial>
      <mass value="3"/>
      <inertia ixx="0.0030645100000000008" ixy="0" ixz="0" iyy="0.0030645100000000008" iyz="0" izz="0.0021774150000000003"/>
    </inertial>
  </link>
  <joint name="right_shoulder_joint" type="revolute">
    <axis rpy="0 0 0" xyz="0 1 0"/>
    <parent link="torso"/>
    <child link="right_shoulder"/>
    <origin rpy="0 0 0" xyz="0 -0.055 0.372"/>
    <limit effort="1.0" lower="0" upper="3.14159" velocity="1.0"/>
  </joint>
  <link name="right_upper_arm_bottom">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://pluto/models/UpperArmBottom.dae" scale="1 1 1"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://pluto/models/UpperArmBottom.dae" scale="1 1 1"/>
      </geometry>
      <surface>
        <friction>
          <ode>
            <mu>0.01</mu>
            <mu2>0.01</mu2>
          </ode>
        </friction>
      </surface>
    </collision>
    <inertial>
      <mass value="0.5"/>
      <inertia ixx="0.002814174479166667" ixy="0" ixz="0" iyy="0.002814174479166667" iyz="0" izz="0.000252015625"/>
    </inertial>
  </link>
  <joint name="right_upper_arm_bottom_joint" type="revolute">
    <axis rpy="0 0 0" xyz="1 0 0"/>
    <parent link="right_shoulder"/>
    <child link="right_upper_arm_bottom"/>
    <origin rpy="0 0 0" xyz="0 -0.0225 0"/>
    <limit effort="1.0" lower="-3.14159" upper="0" velocity="1.0"/>
  </joint>
  <link name="right_upper_arm_top">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://pluto/models/UpperArmTop.dae" scale="1 1 1"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://pluto/models/UpperArmTop.dae" scale="1 1 1"/>
      </geometry>
      <surface>
        <friction>
          <ode>
            <mu>0.01</mu>
            <mu2>0.01</mu2>
          </ode>
        </friction>
      </surface>
    </collision>
    <inertial>
      <mass value="4"/>
      <inertia ixx="0.022513395833333335" ixy="0" ixz="0" iyy="0.022513395833333335" iyz="0" izz="0.002016125"/>
    </inertial>
  </link>
  <joint name="right_upper_arm_top_joint" type="revolute">
    <axis rpy="0 0 0" xyz="0 0 1"/>
    <parent link="right_upper_arm_bottom"/>
    <child link="right_upper_arm_top"/>
    <origin rpy="0 0 0" xyz="0 0 -0.03125"/>
    <limit effort="1.0" lower="0.0" upper="3.14159" velocity="1.0"/>
  </joint>
  <link name="right_forearm_bottom">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://pluto/models/ForearmBottom.dae" scale="1 1 1"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://pluto/models/ForearmBottom.dae" scale="1 1 1"/>
      </geometry>
      <!--surface>
         <friction>
          <ode>
            <mu>1000.0</mu>
            <mu2>1000.0</mu2>
          </ode>
         </friction>
       </surface-->
    </collision>
    <inertial>
      <mass value="0.5"/>
      <inertia ixx="0.0005561144791666667" ixy="0" ixz="0" iyy="0.0005561144791666667" iyz="0" izz="0.000252015625"/>
    </inertial>
  </link>
  <joint name="right_forearm_bottom_joint" type="revolute">
    <axis rpy="0 0 0" xyz="0 1 0"/>
    <parent link="right_upper_arm_top"/>
    <child link="right_forearm_bottom"/>
    <origin rpy="0 1.5707963267948966 0" xyz="0 0 -0.07625"/>
    <limit effort="1.0" lower="0" upper="3.14159" velocity="1.0"/>
    <dyanamics damping="10.0" friction="10.0"/>
  </joint>
  <link name="right_forearm_top">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://pluto/models/ForearmTop.dae" scale="1 1 1"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://pluto/models/ForearmTop.dae" scale="1 1 1"/>
      </geometry>
      <!--surface>
         <friction>
          <ode>
            <mu>1.0</mu>
            <mu2>1.0</mu2>
          </ode>
         </friction>
       </surface-->
    </collision>
    <inertial>
      <mass value="0.4"/>
      <inertia ixx="0.0004448915833333333" ixy="0" ixz="0" iyy="0.0004448915833333333" iyz="0" izz="0.00020161250000000003"/>
    </inertial>
  </link>
  <joint name="right_forearm_top_joint" type="revolute">
    <axis rpy="0 0 0" xyz="1 0 0"/>
    <parent link="right_forearm_bottom"/>
    <child link="right_forearm_top"/>
    <origin rpy="0 0 0" xyz="0.029 0 0"/>
    <limit effort="1.0" lower="0.0" upper="3.14159" velocity="1.0"/>
  </joint>
</robot>

